runtime/onert/core/src/compiler/ExecutorFactory.cc

```diff
@@ -242,6 +291,18 @@ createBackendContexts(compiler::ILoweredGraph &lgraph, bool linear_executor,
                  [&](const auto &ind) { return graph->operations().exist(ind); });
     data.is_linear_executor = linear_executor;
     data.custom_kernel_builder = custom_kernel_builder;
+    if (memory_sharing_supported_backends.count(backend->config()->id()) > 0)
+    {
+      for (const auto &op_ind : op_order)
+      {
+        const auto &op = graph->operations().at(op_ind);
+        if (memory_sharing_allowed(op))
+        {
+          data.shared_memory_operand_map[op.getOutputs().at(0)] = op.getInputs().at(0);
+        }
+      }
+      reassign_indexes_to_single_sources(data.shared_memory_operand_map);
+    }
```

Comment:
I see your point. My intention was to make this mechanism more global but I see that it can be not applicable for other backends.
As you notice it's very problematic to move it to `KernelGenerator` - we need to pass this information for `TensorBuilder` ctor (setter seems to be not good approach and to `initConsts` (here the only possibility seems to be local backend context).
My proposition (already implemented) is to call it in [runtime/onert/backend/cpu/Backend.h](https://github.com/Samsung/ONE/pull/14057/files#diff-9ae39952b733706270d7a4a23e5ae702fb718f11b2048d438923b8ba05dd3eb6)